### PR TITLE
Ensure degree and arcmin symbols are not used when prefixed

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address the fact that as is prefixed degrees and arcminutes use the symbols (as in `m°`). The solution is the same used in #911 (!) for arcseconds -- apparently, it was not noticed at the time that the same should be done for the others. 

This doesn't quite seem worth backporting, but if others disagree, fine to do it.

### CODE REVIEW

File: astropy/units/si.py [hunk 1]
```diff
@@ -109,6 +107,8 @@
     doc="arc second: angular measurement",
 )
 # These special formats should only be used for the non-prefix versions
+deg._format = {"latex": r"{}^{\circ}", "unicode": "°"}
+arcmin._format = {"latex": r"{}^{\prime}", "unicode": "′"}
 arcsec._format = {"latex": r"{}^{\prime\prime}", "unicode": "″"}
```
Comment (olebole) [inline astropy/units/si.py:112]: Having a `._format` additionally to the `.format` for a unit seems a bit hacky to me, and is not straightforward to understand. Sure this was invented long time ago, but shouldn't the attribute name changed here to something more descriptive before it gets extended? Like `.format_simple` or so`

Comment (mhvk) [reply astropy/units/si.py:112]: Definitely hacky, but clearly with precedent... Ideally, we have some kind of hint in `def_unit` that the format should only be used for non-prefix, and for power=1. But maybe best to keep that for another PR...

Comment (nstarman) [reply astropy/units/si.py:112]: Agreed. A refactor / simplification would be good for a followup, but for this PR extending an existing framework is a good solution.

**Review by nstarman — APPROVED** [review]

File: astropy/units/si.py (hunk 1 above)
Comment (olebole) [reply astropy/units/si.py:112]: OK, agreed

**Pull request merged**